accounts/service.go

```diff
@@ -439,15 +439,9 @@ func (s *InterceptorService) AssociateInvoice(ctx context.Context, id AccountID,
 	s.Lock()
 	defer s.Unlock()
 
-	account, err := s.store.Account(ctx, id)
-	if err != nil {
-		return err
-	}
-
-	account.Invoices[hash] = struct{}{}
 	s.invoiceToAccount[hash] = id
 
-	return s.store.UpdateAccount(ctx, account)
+	return s.store.AddAccountInvoice(ctx, id, hash)
```

Comment:
Just noting that this changes the behaviour from the previous code, where we will now assign `s.invoiceToAccount[hash]` the `id` even if an account for that `id` doesn't exist in the db.
So if users are were to manipulate the account id with the context to an unknown id, that would occur.

I do think this should be fine even if that would somehow happen, despite the "memory leak" with the added entry to the map, as it must represent a real invoice which has added a lot of memory allocation either way, and the user wouldn't get credited if it was paid.

Reply:
good catch! I've updated it to only assign to the in-mem map if the DB call succeeded. lemme know what you think